[Doc] add notes on specialties on py-spy stack traces (#43509)
ray-project/ray: doc/source/ray-observability/user-guides/debug-apps/debug-hangs.rst

   * if you start Ray manually in a Docker container, follow the `py-spy documentation`_ to resolve it. 
   * if you are a KubeRay user, follow the :ref:`guide to configure KubeRay <kuberay-pyspy-integration>` and resolve it.
   
.. note::

>    The following errors are conditional and not signals of failrues for your Python programs:

Reviewer: ```suggestion
   The following errors are conditional and not signals of failures for your Python programs:
```

   * If you see "No such file or direction", check if your worker process have exited.

>    * If you see "No stack counts found", check if your worker may be sleeping and not active in the last 5s.

Reviewer 2: ```suggestion
   * If you see "No stack counts found", check if your worker process was sleeping and not active in the last 5s.
```
